File: amazon/1_two_sum.cpp

class Solution{
public:
    vector<int> twoSum(vector<int>& nums, int target){
        unordered_map<int, int> m1;
        vector<int> ans;
        for(int i = 0; i < nums.size(); i++){
            if(m1.count(target-nums[i])){
                ans.push_back(m1[target-nums[i]]);
                ans.push_back(i);
                return ans;
            }
            m1[nums[i]] = i;
        }
        return nums;
    }
};

File: amazon/1_two_sum.py

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numDict = {}
        n = len(nums)
        for i in range(n) :
            result = target - nums[i]
            if result in numDict :
                return [numDict[result], i]
            if nums[i] not in numDict :
                numDict[nums[i]] = i
        return []